resources/views/courses/action.blade.php

```diff
@@ -2,7 +2,7 @@
     @if (Auth::user()->student->isEnrolledInCourse($course))
         <button type="button" class="btn btn-outline-secondary btn-sm disabled">Enrolled</button>
 
-        @if ($settings->withinEnrollmentPeriod())
+        @if ($settings->withinEnrollmentPeriod() && is_null(Auth::student()->get()->getEnrollmentWithShiftByCourse($course)))
```

Comment:
I can change this to a method into a controller, but the information is from two different *tables*, **Settings** and **Enrollments**, so I don't think we can make it a single query execution. Just tell me what you think.

Reply:
Settings is not a problem, since it is fetched at the beginning of the lifecycle of the application.

The problem is to execute `getEnrollmentWithShiftByCourse` several times.

See the `scopeOrderByCourse`, in Enrollment model. Despite being a bit different, I suggest you to make some experiments with Eloquent ORM.